### DISCUSSION TITLE
Add line breaks for formatting purposes

### DISCUSSION THREAD

**Discussion by SilasRodrigues19** [alpinejs/alpine#3835] [1. Help] [answered]
I'm using Alpine with Astro and Tailwind, it turns out that when I try to add line breaks to make the code more readable, VSCode starts displaying ts errors indicating that the string is not finalized.

`Unterminated string literal. ts(1002)`

When I do this, my code continues to work, but the file turns red in VSCode signaling an error. Is there any extension or any other way so that I can use line breaks without having errors in the file?

Example:

The :class line is too long

```astro
<header
  @click="open ? window.scrollTo(0, 0) : null"
  x-data="{ open: false, windowPosition: 0 }"
  x-init="() => { window.addEventListener('scroll', () => { windowPosition = window.scrollY; }); }"
  class="fixed z-10 w-full transition"
  :class="{ 'bg-transparent dark:bg-transparent': !(windowPosition > 0) && (!open), 'bg-primary-500/10 dark:bg-primary-400/10 relative h-auto': open, 'bg-primary-50 dark:bg-primary-950': !(open) }"
  aria-labelledby="header-navigation"
>
```

I would like to break lines like this:

```astro
<header
  @click="open ? window.scrollTo(0, 0) : null"
  x-data="{ open: false, windowPosition: 0 }"
  x-init="() => { window.addEventListener('scroll', () => { windowPosition = window.scrollY; }); }"
  class="fixed z-10 w-full transition"
  :class="{ 
    'bg-transparent dark:bg-transparent': !(windowPosition > 0) && (!open), 
    'bg-primary-500/10 dark:bg-primary-400/10 relative h-auto': open, 
    'bg-primary-50 dark:bg-primary-950': !(open) 
  }"
  aria-labelledby="header-navigation"
>
```

This way, no errors appear and the file does not turn red signaling errors:

![image](https://github.com/alpinejs/alpine/assets/47532703/5e8fd559-6463-41b6-a620-af22f83a79d2)

This way, the code becomes more readable but the mentioned error starts to appear and the file is highlighted in red (I would like some way to be able to break lines like this):

![image](https://github.com/alpinejs/alpine/assets/47532703/29e99124-9d27-4169-a3eb-8ca97e7863e8)

**Comment by ekwoka** [comment, marked as answer]
This is an astro thing, not an alpine thing.

Basically the system you are using that is producing the error, seems like TS, is evaluating it as JS, in which `"` and a new line is an error.

idk how exactly astro works, but maybe it will work fine with backticks (that would make TS happy), which do allow new lines and whitespace.

**Comment by SilasRodrigues19** [comment]
Hey @ekwoka  

I confess that I hadn't even remembered the backtick, I think it's worth using backticks to have better readability in this case.

If I find out the real reason I'll comment here again, but the crasis solves my problem. Thanks
